### ISSUE TITLE
Full Support for DLC Content

### ISSUE THREAD

**Issue by mpstark** [BattletechModders/ModTek#90]
* Provide `RequiresDLC` for both ModDef and ModEntry
* Defer handling merges for DLC content until unpacked and available

**Comment by CptMoore** [comment]
I really need this to fix the cyclops, it currently contains invalid hardpoint data

```
Error: AdvancedJSONMerge: "MechEngineer\advancedjsonmerge\hardpointdatadef_cyclops.json" could not resolve type for ID: hardpointdatadef_cyclops. Skipping this merge
```

**Comment by mcb5637** [comment]
I did some testing with BT and Unitys AssetBundles.
Turns out, they are not loaded while ModTek makes its merges, so in order to get something out of them, they would have to be loaded, then accessed and finally unloaded.
Quick example:
```cs
            AssetBundle a = AssetBundle.LoadFromFile(path+"BattleTech_Data\\StreamingAssets\\data\\assetbundles\\heavymetal");
            FileLog.Log("found in " + a.name + " " + a.Contains("Gear_JumpJet_Vectored_Thrust_Kit"));
            FileLog.Log(a.LoadAsset<TextAsset>("Gear_JumpJet_Vectored_Thrust_Kit").text);
            a.Unload(true);
```
This was executed as a ModTek dll in FinishedLoading, so it should be relatively easy to adapt this to checks if some json is in one of the dlc assetbundles, and if the check is true, load the json as a string to perform the merge.
I would like to test this out, but it is probably faster if someone that knows the code helps me out by pointing me to the correct code pieces. What i need is:
- A method that gets called before the merge (to load the assetbundles)
- The path to the BT install location
- A method that checks if a file to merge exists
- A method that loads a file to merge
- A method that gets called after the merges are completed (to unload the assetbundles)
